components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/processor/JWTInboundRequestProcessor.java

```diff
@@ -0,0 +1,669 @@
+                    log.debug("Session not exists. Sending request to identity framework for authentication.");
+                }
+                return buildResponseForFrameworkLogin(messageContext);
+            }
+        } else {
+            String msg = "A Service Provider with the Relying Party '" + neutralize(this.relyingParty) + "' is not " +
+                    "registered. Service Provider should be registered in advance.";
+            log.error(msg, new JWTIdentityException(msg));
+            return JWTInboundUtil.sendToRetryPage(JWTInboundConstants.ErrorMessages.MISCONFIGURATION_STATUS,
+                    JWTInboundConstants.ErrorMessages.MISCONFIGURATION_MESSAGE);
+        }
+    }
+
+    /**
+     * Validates the endpoint URL
+     *
+     * @param identityRequest The identity request
+     * @return
+     */
+    private boolean validateEndpointUrl(IdentityRequest identityRequest) {
+
+        this.endpointUrl = getPropertyValue(identityRequest, JWTInboundConstants.SPBasedConfigs.SITE_API_URL);
+        if (StringUtils.isNotBlank(this.endpointUrl)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Setting the endpoint URL: " + neutralize(this.endpointUrl));
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Validates the JWS algorithm
+     *
+     * @param identityRequest The identity request
+     * @return
+     */
+    private boolean validateJwsAlgorithm(IdentityRequest identityRequest) {
+
+        setJWSAlgorithm(getPropertyValue(identityRequest, JWTInboundConstants.SPBasedConfigs.JWS_ALGORITHM));
+        if (this.jwsAlgorithm != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Setting the JWS Algorithm");
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Setting the redirect URL parameter and validating against the regex configured in the SP configuration.
+     *
+     * @param identityRequest The identity request
+     * @return True if the redirect URL is valid or not provided. False if the validation failed against the regex.
+     */
+    private boolean handleRedirectUrl(IdentityRequest identityRequest) {
+
+        // Set Redirect URL
+        String redirectUrlParamName = JWTInboundConstants.SPDefaultValueConfigs.REDIRECT_URL_DEFAULT_PARAM_VALUE;
+        if (StringUtils.isNotBlank(getPropertyValue(identityRequest, JWTInboundConstants.SPBasedConfigs.
+                REDIRECT_URL_PARAM_NAME))) {
+            // Get the redirect URL parameter name from the SP config if it is configured
+            redirectUrlParamName = getPropertyValue(identityRequest, JWTInboundConstants.SPBasedConfigs.
+                    REDIRECT_URL_PARAM_NAME);
+        }
+        this.redirectUrl = identityRequest.getParameter(redirectUrlParamName);
+        if (log.isDebugEnabled()) {
+            log.debug("Redirect URL parameter name: " + neutralize(redirectUrlParamName) +
```

Comment:
In the log can we say for which RP as well

Reply:
Fixed